sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -132,19 +140,20 @@ def test_mutual_info_regression():
     mean = np.zeros(4)
 
     rng = check_random_state(0)
-    Z = rng.multivariate_normal(mean, cov, size=1000)
+    Z = rng.multivariate_normal(mean, cov, size=1000).astype(dtype)
     X = Z[:, 1:]
     y = Z[:, 0]
 
     mi = mutual_info_regression(X, y, random_state=0)
     assert_array_equal(np.argsort(-mi), np.array([1, 2, 0]))
```

Comment:
Could you please add an assertion on the dtype of `mi`?

It seems not to be the case because, but by reading this code, it does not seem to be intentional as I see several lines of code in `_estimate_mi` that are likely to introduce an upcasting.

Maybe we can add an assertion such as:

```suggestion
    assert_array_equal(np.argsort(-mi), np.array([1, 2, 0]))
    # XXX: should mutual_info_regression be fixed to avoid
    # up-casting float32 inputs to float64?
    assert mi.dtype == np.float64
```